clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsClusterProvider.java

```diff
@@ -29,29 +29,25 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
-import static com.netflix.spinnaker.clouddriver.ecs.cache.Keys.Namespace.ECS_CLUSTERS;
-import static com.netflix.spinnaker.clouddriver.ecs.cache.Keys.SEPARATOR;
-
 @Component
 public class EcsClusterProvider {
 
-  private Cache cacheView;
+  private EcsClusterCacheClient ecsClusterCacheClient;
 
   @Autowired
   public EcsClusterProvider(Cache cacheView) {
-    this.cacheView = cacheView;
+    this.ecsClusterCacheClient = new EcsClusterCacheClient(cacheView);
   }
 
   public List<String> getEcsClusters(String account, String region) {
-    Collection<String> ecsClustersCache = cacheView.filterIdentifiers(ECS_CLUSTERS.toString(),
-      "ecs" + SEPARATOR + ECS_CLUSTERS + SEPARATOR + account + SEPARATOR + region + SEPARATOR + "*");
+    Collection<EcsCluster> ecsClusters = ecsClusterCacheClient.getAll(account, region);
```

Comment:
I'd make `ecsClusterCacheClient.getAll(...)` return an empty collection rather than null, which would simplify the code below and make it safer to use

Reply:
Done.